cargo-public-api/src/published_crate.rs

```diff
@@ -0,0 +1,113 @@
+//! Creates a dummy project with a dependency on the crate we want to build
+//! rustdoc JSON for. We then build rustdoc JSON for the crate using this dummy
+//! project.
+
+use std::{fmt::Display, path::PathBuf};
+
+use anyhow::{anyhow, Result};
+
+use crate::Args;
+
+#[derive(Debug, PartialEq, Eq)]
+struct PackageSpec {
+    name: String,
+    version: String,
+}
+
+impl PackageSpec {
+    fn as_dir_name(&self) -> PathBuf {
+        PathBuf::from(format!("{}-{}", self.name, self.version))
+    }
+}
+
+impl Display for PackageSpec {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}@{}", &self.name, &self.version)
+    }
+}
+
+impl TryFrom<&str> for PackageSpec {
+    type Error = anyhow::Error;
+
+    fn try_from(spec_str: &str) -> Result<Self, Self::Error> {
+        let mut split = spec_str.split('@');
+        let name = split.next().map(str::to_owned);
+        let version = split.next().map(str::to_owned);
+
+        match (name, version) {
+            (Some(name), Some(version)) if !name.is_empty() && !version.is_empty() => Ok(Self {
+                name,
+                version,
+            }),
+            _ => Err(anyhow!("Invalid format of package spec string. Use `crate-name@version`, e.g. `rustdoc-json@0.4.0`")),
+        }
+    }
+}
+
+pub fn build_rustdoc_json(package_spec_str: &str, args: &Args) -> Result<PathBuf> {
+    let spec = PackageSpec::try_from(package_spec_str)?;
+
+    // We don't need to preserve tempdir because we just want the cached crate in $HOME
+    let mut temp_package = build_dir_root();
+    temp_package.push("cargo-public-api_build-root-for-published-crates");
```

Comment:
```suggestion
    temp_package.push("cargo-public-api");
```

Reply:
Thanks for taking a look at the code; always appreciated. May I ask why you want to change to just `cargo-public-api `? My reasoning for using `cargo-public-api_build-root-for-published-crates` is that it makes it much easier for users to find the project AND the code (via git grep) that is responsible for putting stuff in their `~/.cargo` dir. The risk is high they will discover that dir out-of-context, i.e. long after having used this feature.

Reply:
my thinking was similar to how `cargo-audit` works, but maybe

```suggestion
    temp_package.join("cargo-public-api/build-root-for-published-crates");
```

then, I agree with the reasoning, just not the long dir name :)

Reply:
Great idea! Let's use that 👍